app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt

```diff
@@ -1,13 +1,97 @@
 package ar.com.wolox.android.example.ui.home.news
 
+import android.annotation.TargetApi
+import android.os.Build
+import ar.com.wolox.android.example.network.News
 import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices
+import java.time.LocalDateTime
 import javax.inject.Inject
 
-class NewsPresenter @Inject constructor() : BasePresenter<INewsView>() {
+@TargetApi(Build.VERSION_CODES.O)
+class NewsPresenter @Inject constructor(retrofitServices: RetrofitServices) : BasePresenter<INewsView>() {
+
+    private val mNewsAdapterAPI = NewsAdapterAPI(retrofitServices)
 
     override fun onViewAttached() {}
 
     fun onAddNewsButtonPressed() = view.goAddNews()
 
-    fun onPullDownRefresh() = view.nothingNewToShow()
+    /**
+     * This function doesn't compare
+     * older news with recent ones,
+     * for now.
+     */
+    fun onLoadOldNews() {
+        mNewsAdapterAPI.loadOlderNews({ onSuccessOlderNews(newsList = it) }, { onEmptyList() }, { onFailureOlderNews() })
+    }
+
+    private fun onSuccessOlderNews(newsList: ArrayList<News>) {
+        view.addOlderNews(newsList.apply { setReadableCreationTime(this) })
+    }
+
+    private fun onEmptyList() {
+        view.completeLoading()
+        view.nothingNewToShow()
+    }
+
+    private fun onFailureOlderNews() {
+        view.onLoadOlderNewsError()
+    }
+
+    fun onLoadRecentNews() {
+        view.startLoading()
+        mNewsAdapterAPI.loadRecentNews({ onSuccessRecentNews(newsList = it) }, { onEmptyList() }, { onFailureRecentNews() })
+    }
+
+    private fun onSuccessRecentNews(newsList: ArrayList<News>) {
+        view.completeLoading()
+        newsList.apply {
+            sortByDescending { it.getCreatedAt() }
+            setReadableCreationTime(this)
+        }
+        view.addRecentNews(newsList)
+    }
+
+    private fun onFailureRecentNews() {
+        view.completeLoading()
+        view.onLoadRecentNewsError()
+    }
+
+    private fun setReadableCreationTime(newsList: ArrayList<News>): ArrayList<News> {
+        val now = LocalDateTime.now()
+
+        for (i in 0..newsList.lastIndex) {
+            var yearsOfDifference = newsList[i].getCreatedAt().year - now.year
+            val monthOfTheNewIsNow = newsList[i].getCreatedAt().month == now.month
+
+            if (yearsOfDifference == 0) {
```

Comment:
This would be muuuuch simpler and legible using `when` expression.
```
readableTime = when {
    now.year > createdAt.year  -> "${yearsOfDifference * -1} years"
    now.month > createdAt.month -> ...
    now.dayOfMonth > createdAt.dayOfMonth -> ...
    now.hour > createdAt.hour -> ...
    ...
}
```